sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1855,24 +1855,22 @@ device_image_plain ProgramManager::build(const device_image_plain &DeviceImage,
     auto [NativePrg, DeviceCodeWasInCache] = getOrCreatePIProgram(
         Img, Context, Devs[0], CompileOpts + LinkOpts, SpecConsts);
 
-    if (!DeviceCodeWasInCache) {
-      if (InputImpl->get_bin_image_ref()->supportsSpecConstants())
-        enableITTAnnotationsIfNeeded(NativePrg, Plugin);
-
-      {
-        std::lock_guard<std::mutex> Lock{InputImpl->get_spec_const_data_lock()};
-        const std::map<std::string,
-                       std::vector<device_image_impl::SpecConstDescT>>
-            &SpecConstData = InputImpl->get_spec_const_data_ref();
-
-        for (const auto &DescPair : SpecConstData) {
-          for (const device_image_impl::SpecConstDescT &SpecIDDesc :
-               DescPair.second) {
-            if (SpecIDDesc.IsSet) {
-              Plugin.call<PiApiKind::piextProgramSetSpecializationConstant>(
-                  NativePrg, SpecIDDesc.ID, SpecIDDesc.Size,
-                  SpecConsts.data() + SpecIDDesc.BlobOffset);
-            }
+    if (!DeviceCodeWasInCache &&
```

Comment:
I do not see any changes (other than some format changes) here. Can we do away with this change? Thanks

Reply:
There is a change, the whole block is now conditioned on `InputImpl->get_bin_image_ref()->supportsSpecConstants()` when before it was only the call to `enableITTAnnotationsIfNeeded`, this is necessary because `piextProgramSetSpecializationConstant` is only supposed to be called for native specialization constants.

It's a little hard to see because of the way it was written originally, "one-line" if with no brackets followed by a scope.

Reply:
Also note that this is part of https://github.com/intel/llvm/pull/6125 , maybe I should fuse the two MRs.